Sources/InfomaniakCore/Calendar/RruleDecoder.swift

```diff
@@ -0,0 +1,272 @@
+//
+//  RruleDecoder.swift
+//  InfomaniakCore
+//
+//  Created by Baptiste on 25.02.2025.
+//
+
+import Foundation
+
+public struct RruleDecoder: Sendable {
+    public enum Frequency: String, CaseIterable, Codable, Sendable {
+        case secondly = "SECONDLY"
+        case minutely = "MINUTELY"
+        case hourly = "HOURLY"
+        case daily = "DAILY"
+        case weekly = "WEEKLY"
+        case monthly = "MONTHLY"
+        case yearly = "YEARLY"
+    }
+
+    public enum Weekday: String, CaseIterable, Codable, Sendable {
+        case monday = "MO"
+        case tuesday = "TU"
+        case wednesday = "WE"
+        case thursday = "TH"
+        case friday = "FR"
+        case saturday = "SA"
+        case sunday = "SU"
+    }
+
+    public enum rule: String, CaseIterable, Codable {
+        case frequency = "FREQ"
+        case interval = "INTERVAL"
+        case count = "COUNT"
+        case until = "UNTIL"
+        case byDay = "BYDAY"
+    }
+
+    public let frequency: Frequency
+    public let calendar: Calendar
+    public let interval: Int?
+    public let end: Int?
+    public let count: Int?
+    public let byDay: [Weekday]?
+
+    public init(frequency: Frequency, interval: Int?, calendar: Calendar = .current, end: Int?, count: Int?, byDay: [Weekday]?) {
+        self.frequency = frequency
+        self.interval = interval
+        self.calendar = calendar
+        self.end = end
+        self.count = count
+        self.byDay = byDay
+    }
+}
+
+// MARK: - ParseStrategy
+
+extension RruleDecoder: ParseStrategy {
+    public enum DomainError: Error {
+        case invalidInterval
+        case invalidKey
+        case invalidCount
+        case invalidUntil
+        case invalidByDay
+        case missingFrequency
+        case bothUntilAndCountSet
+    }
+
+    public func parse(_ value: String) throws -> RruleDecoder {
+        var frequency: Frequency?
+        var interval: Int?
+        var count: Int?
+        var end: Int?
+        var countOrUntilSet = 0
+        var byDay: [RruleDecoder.Weekday] = []
+
+        let parts = value.split(separator: ";")
+
+        for part in parts {
+            let keyValue = part.split(separator: "=")
+            guard keyValue.count == 2 else {
+                throw DomainError.invalidKey
+            }
+
+            let key = keyValue[0]
+            let val = String(keyValue[1])
+
+            switch key {
```

Comment:
You can create a protocol like "ItemDecoder" (hope you will come with a better name than me)
Then, you can create a struct to decode a Frequency/Interval… that conforms to ItemDecoder.
In your switch, you can then do something like `frequency = key.decoder.decode(value)`